File: C#/Solutions/Problem015/Problem015.cs
using System;
using System.Collections.Generic;
using Xunit;

namespace DailyCodingProblems.Solutions.Problem015
{
    /// <summary>
    /// This problem was asked by Facebook.
    /// 
    /// Given a stream of elements too large to store in memory, pick a random element from the stream with uniform probability.
    /// </summary>
    public class Problem015
    {
        [Fact]
        public void Problem015SolutionTest()
        {
            var stream = new List<int> {1, 2, 3, 4, 5, 6, 7, 8, 9};

            Assert.Contains(Stream<int>.GetRandomElementFromStream(stream), stream);
        }
    }
}

File: Java/Solutions/Problem015/Problem015.java
package Problem015;

import org.junit.Assert;
import org.junit.Test;

import java.util.*;

/**
 * This problem was asked by Facebook.
 *
 * Given a stream of elements too large to store in memory, pick a random element from the stream with uniform probability.
 */
public class Problem015 {

    @Test
    public void problem015SolutionTest() {
        var stream = Arrays.asList(1, 2, 3, 4, 5, 6, 7, 8, 9);
        Assert.assertTrue(stream.contains(Stream.GetRandomElementFromStream(stream)));
    }
}

